client/src/components/QA/AddingForm.jsx

```diff
@@ -80,8 +79,9 @@ class AddingForm extends React.Component {
     if (this.checkingRequire()) {
       axios.post(url, data)
         .then(res => {
-          console.log(res);
-          this.props.closePopup();
+          console.log('submit success', res);
```

Comment:
The only thing that stands out to me here, which I should have noticed in a prior PR, is that you're binding functions in your render method. Standard practice in React is to do this in your constructor: `this.handleOnChange = this.handleOnChange.bind(this);` Then you can just put `this.handleOnChange` in your render method and it won't create a new instance of your event handler every time your component renders.